respa_admin/views/resources.py

```diff
@@ -432,6 +436,35 @@ def _save_resource_purposes(self):
         for purpose in checked_purposes:
             self.object.purposes.add(purpose)
 
+    def _save_or_update_opening_hours_via_period_templates(self):
+        checked_period_templates_ids = self.request.POST.getlist('period_templates')
+        checked_period_templates = Period.objects.filter(
+            id__in=[int(p) for p in checked_period_templates_ids]
+        )
+        current_period_from_templates = self.object.periods.filter(template_src__isnull=False)
+        used_templates = [template.template_src for template in current_period_from_templates]
+
+        for current_temp in current_period_from_templates:
+            if current_temp.template_src not in checked_period_templates:
+                self.object.periods.remove(current_temp)
+
+        for selected_period in checked_period_templates:
+            if selected_period in used_templates:
+                continue
+
+            template_id = selected_period.pk
+            days = Day.objects.filter(period=selected_period)
+            selected_period.template_src_id = template_id
+            selected_period.pk = None
+            selected_period.is_template = False
+            selected_period.resource = self.object
+            selected_period.save()
+
+            for day in days:
+                day.pk = None
+                day.period = selected_period
+                day.save()
```

Comment:
It might be just me, but I think this function is pretty hard to understand, so it would be nice if it had some comments on what exactly is happening in different stages of it 🙂

Reply:
Added some comments now :)